File: middleware/src/dsl/Coordinate.java
package dsl;

import java.util.List;
import java.util.Optional;

public class Coordinate {
	private double x;
	private double y;
	private double z;
	
	public Coordinate (double x, double y) {
		this.x = x;
		this.y = y;
	}
	
	public Coordinate (double x, double y, double z) {
		this.x = x;
		this.y = y;
		this.z = z;
	}
	
	public double getX() {
		return x;
	}
	
	public double getY() {
		return y;
	}
	
	public double getZ() {
		return z;
	}
	
	public double distanceSqr(Coordinate c) {
		return Math.pow((c.x - this.x),2.0) + Math.pow((c.y - this.y), 2.0);
	}
	
	public double distance(Coordinate c) {
		return Math.sqrt(distanceSqr(c))
	}
	
	public Optional<Coordinate> findClosestPoint(List<Coordinate> coors) {
		Optional<Coordinate> closest = Optional.empty();
		double distance = Double.MAX_VALUE;
		for (Coordinate c : coors) {
			double d = distance(c);
			if (d < distance) {
				closest = Optional.of(c);
				distance = d;
			}
		}
		return closest;
	}
}

File: middleware/src/dsl/mission/Coordinate.py
import math

class Coordinate:
	x = 0
	y = 0
	z = 0
	
	def __init__(self, x, y):
		self.x = x
		self.y = y
	
	def __init__(self, x, y, z):
		self.x = x
		self.y = y
		self.z = z
	
	def getX(self):
		return self.x
	
	def getY(self):
		return self.y
	
	def getZ(self):
		return self.z
	
	def distanceSqr(self, c):
		return math.pow((c.x - self.x),2.0) + math.pow((c.y - self.y), 2.0)
	
	def distance(self, c):
		return math.sqrt(self.distanceSqr(c))
	
	def findClosestPoint(self, coors):
		closest = []
		distance = math.MAX_VALUE
		for c in coors:
			d = self.distance(c)
			if d < distance:
				closest = c
				distance = d
		return closest